src/main/java/org/broadinstitute/hellbender/engine/ReferenceDataSource.java

```diff
@@ -33,39 +27,17 @@
      *
      * @param fastaFile reference fasta file
      */
-    public ReferenceDataSource( final File fastaFile ) {
-        if ( fastaFile == null )
-            throw new IllegalArgumentException("fastaFile must be non-null");
-
-        // Will throw a UserException if the .fai and/or .dict are missing
-        reference = CachingIndexedFastaSequenceFile.checkAndCreate(fastaFile);
+    public static ReferenceDataSourceFromFile of(final File fastaFile) {
```

Comment:
Also mention in docs for this factory method that "The provided fasta file must have companion .fai and .dict files."